js/intro.js

```diff
@@ -0,0 +1,320 @@
+import { BASE_URL } from './constants.js';
+
+const notAuthorized = document.querySelector('.not-authorized-page');
+const notFound = document.querySelector('.not-found-page');
+const mainContainer = document.querySelector('.intro-main');
+const loading = document.getElementById('loading');
+
+async function makeApiCall(
+  url,
+  method = 'get',
+  body = null,
+  credentials = 'include',
+  headers = { 'content-type': 'application/json' },
+  options = null,
+) {
+  try {
+    const response = await fetch(url, {
+      method,
+      body,
+      headers,
+      credentials,
+      ...options,
+    });
+    return response;
```

Comment:
Convert the response to json format from here and then return.

```js
await response.toJson()
```

Reply:
> Convert the response to json format from here and then return.
> 
> ```js
> await response.toJson()
> ```

not able to do because i am using response.status in code and makeApiCall() fetch my request

https://github.com/Real-Dev-Squad/website-www/pull/432/files/f6c2354ca25c52308fa7baf469bbba3e4bb46107#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R258

Reply:
Use 
```js
await response.json();
```
instead of below `toJson()` method does not exist in Javascript
```js
await response.toJson()
```

Reply:
> > Convert the response to json format from here and then return.
> > ```js
> > await response.toJson()
> > ```
> 
> not able to do because i am using response.status in code and makeApiCall() fetch my request
> 
> https://github.com/Real-Dev-Squad/website-www/pull/432/files/f6c2354ca25c52308fa7baf469bbba3e4bb46107#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R258

You can send response as object/record having following properties:

```jsx
{
data: any,
error: message<string>,
code: number
}
```